### PR TITLE
installed libraries for eslint (airbnb config), prettier, and git hooks

### PR DESCRIPTION
**Pull request by joshflchan**
- added eslint and prettier libraries with airbnb configs
    - disable eslint formatting and use prettier strictly for formatting and eslint for code quality
- added prettier rules
    - please check if sufficient for now

- added husky for easy git hooks and lint-staged libraries to force linting on commit to prevent bad commits

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -18,10 +18,32 @@
...
+      "prettier"
+    ],
+    "rules": {
+      "prettier/prettier": "error",
+      "import/first": "error"
+    }
+  },
+  "prettier": {
+    "printWidth": 90,
+    "bracketSpacing": false,
```
Comment (ianmah) [inline package.json]: kinda want this `true`

File: package.json [hunk 2]
```diff
@@ -34,5 +56,31 @@
       "last 1 firefox version",
       "last 1 safari version"
     ]
+  },
+  "devDependencies": {
+    "eslint": "^6.1.0",
+    "eslint-config-airbnb": "^18.0.1",
+    "eslint-config-prettier": "^6.5.0",
+    "eslint-plugin-import": "^2.18.2",
+    "eslint-plugin-jsx-a11y": "^6.2.3",
```
Comment (ianmah) [inline package.json]: is this part of one of the plugins?

File: package.json [hunk 3]
```diff
@@ -34,5 +56,31 @@
...
+    "eslint-plugin-prettier": "^3.1.1",
+    "eslint-plugin-react": "^7.14.3",
+    "eslint-plugin-react-hooks": "^1.7.0",
+    "prettier": "^1.19.1"
+  },
+  "lint-staged": {
```
Comment (ianmah) [inline package.json]: How does this work - when do these commands run, at `git commit` or `git add`?

File: package.json [hunk 4]
```diff
@@ -18,10 +18,32 @@
...
+      "react-app",
+      "airbnb",
+      "prettier"
+    ],
+    "plugins": [
+      "prettier"
+    ],
+    "rules": {
+      "prettier/prettier": "error",
+      "import/first": "error"
```
Comment (ianmah) [inline package.json]: what does this do?

File: package.json [hunk 5]
```diff
@@ -18,10 +18,32 @@
     "build": "react-scripts build",
     "deploy": "aws s3 sync build/ s3://biztech-web",
     "test": "react-scripts test",
-    "eject": "react-scripts eject"
+    "eject": "react-scripts eject",
+    "lint": "eslint .",
+    "format": "prettier --write \"**/*.+(js|jsx|json|yml|yaml|css|md|vue)\""
```
Comment (ianmah) [inline package.json]: are these scripts ever run automatically? Or do we have to run them ourselves

File: package.json (hunk 2 above)
Comment (joshflchan) [reply package.json]: part of eslint

File: package.json (hunk 3 above)
Comment (joshflchan) [reply package.json]: when staged, so should be on `git add`

File: package.json (hunk 5 above)
Comment (joshflchan) [reply package.json]: for manual running:
- `npm run lint` uses eslint to check code-quality of all files
- `npm run format` uses prettier to format all files that end with the listed suffixes

will note in mention

File: package.json (hunk 4 above)
Comment (joshflchan) [reply package.json]: - disables eslint formatting and uses prettier formatting
- produces error if import statements are below other parts of code

**Comment by joshflchan** [comment]
Currently WIP, attempting to fix all ESLint errors before pushing
The long I wait, the more code there will be to be fixed 😢

**Pull request closed without merging**